stf/src/saros/stf/server/rmi/superbot/impl/SuperBot.java

```diff
@@ -78,26 +75,17 @@ public void confirmShellAddProjectWithNewProject(String projectName) throws Remo
   public void confirmShellAddProjectUsingExistProject(String projectName) throws RemoteException {
     SWTBot bot = new SWTBot();
     bot.waitUntil(
-        Conditions.shellIsActive(SHELL_ADD_PROJECTS), SarosSWTBotPreferences.SAROS_LONG_TIMEOUT);
+        Conditions.shellIsActive(SHELL_ADD_RESOURCES), SarosSWTBotPreferences.SAROS_LONG_TIMEOUT);
 
-    SWTBotShell shell = bot.shell(SHELL_ADD_PROJECTS);
+    SWTBotShell shell = bot.shell(SHELL_ADD_RESOURCES);
     shell.activate();
 
-    // see https://bugs.eclipse.org/bugs/show_bug.cgi?id=344484
-    // FIXME fix the bug in the SWTBot Framework, do not use a workaround
-    final Button radioButton = shell.bot().radio(RADIO_CREATE_NEW_PROJECT).widget;
+    shell.bot().radio(RADIO_USING_EXISTING_DIRECTORY).click();
 
-    UIThreadRunnable.syncExec(
-        new VoidResult() {
+    // Ensure that dialog had enough time to process radio button switch
+    bot.sleep(1000);
```

Comment:
Why ? Is there some "magic involved" ?! You only (and never) need to sleep to when something is done in a background thread that will change the UI in the "near" future.

Reply:
I am not really certain on why this helped.

The issue with the test as it was before is that the STF fails while trying to write to the text filed for the existing directory path, stating that it is "read only". When looking at the logic, it seems like [this check](https://github.com/eclipse/swtbot/blob/master/org.eclipse.swtbot.swt.finder/src/org/eclipse/swtbot/swt/finder/widgets/SWTBotText.java#L67) fails while trying to set the text.

The only way I could make sense of it is that the field is still disabled and is therefore seen as "read only". But, having a closer look at it now, the text box being disabled should lead to a different error message. So I have no idea what the problem is.

Either way, with the delay, the error message went away. I did not change anything else. But I am also not sure what else could be the issue. I am not setting the text boxes to non-editable in the logic at any point.

Reply:
I only remember the bug where selecting radio buttons will not fire listener events. Is this fixed in our SWTBot version ?

Reply:
Instead of a stupid wait you could also wait for the field to become writeable. This is a lot "faster" as the check is done every xyz milliseconds (can be globally configured)

Reply:
For whatever reason, I am no longer able to reproduce the issue locally (even though I don't think I changed anything in the related UI logic), so I removed the sleep for now. As already mentioned, if the issue crops up again, I will add the conditional wait.

But for now, I have to figure out why so many tests still fail on the build server that pass locally.